fetch-in-react-practice/client/components/app.jsx

```diff
@@ -0,0 +1,103 @@
+import React from 'react';
+import PageTitle from './page-title';
+import TodoList from './todo-list';
+import TodoForm from './todo-form';
+
+export default class App extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      todos: []
+    };
+    this.addTodo = this.addTodo.bind(this);
+    this.toggleCompleted = this.toggleCompleted.bind(this);
+  }
+
+  componentDidMount() {
+    this.getAllTodos();
+  }
+
+  getAllTodos() {
+    /**
+     * Use fetch to send a GET request to `/api/todos`.
+     * Then 😉, once the response JSON is received and parsed,
+     * update state with the received todos.
+     */
+    fetch('https://localhost:3000/api/todos', {
+      method: 'GET'
+    })
+      .then(res => res.json())
+      .then(data => this.setState({ todos: data }))
+      .catch(err => console.error('getAllTodos failed!', err));
+  }
+
+  addTodo(newTodo) {
+    /**
+    * Use fetch to send a POST request to `/api/todos`.
+    * Then 😉, once the response JSON is received and parsed,
+    * add the created todo to the state array.
+    *
+    * TIP: Be sure to SERIALIZE the todo object in the body with JSON.stringify()
+    * and specify the "Content-Type" header as "application/json"
+    */
+    fetch('https://localhost:3000/api/todos', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(newTodo),
+      data: {
+        task: newTodo.task,
+        isCompleted: false
+      }
+    })
+      .then(res => res.json())
+      .then(data => this.state.todos.push(newTodo))
```

Comment:
Here you can't directly push the new todo to state. You need to copy the todos array, add the new todo item, and then setState.